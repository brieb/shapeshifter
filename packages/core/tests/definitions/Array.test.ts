import { options } from '../mocks';
import ArrayDefinition from '../../src/definitions/Array';
import StringDefinition from '../../src/definitions/String';

describe('definitions/Array', () => {
  it('errors if `valueType` is missing', () => {
    expect(() => new ArrayDefinition(options, 'foo', {})).toThrow(
      'Invalid ArrayDefinition field "valueType". Field is required and must be defined.',
    );
  });

  it('errors if `valueType` is empty', () => {
    expect(
      () => new ArrayDefinition(options, 'foo', { valueType: '' }),
    ).toThrowErrorMatchingSnapshot();
  });

  it('errors if `valueType` is the wrong type', () => {
    // @ts-ignore
    expect(() => new ArrayDefinition(options, 'foo', { valueType: 123 })).toThrow(
      'Invalid ArrayDefinition field "valueType". Type must be one of: string, shape',
    );
  });

  it('creates a `Definition` for the `valueType`', () => {
    const def = new ArrayDefinition(options, 'foo', { valueType: 'string' });

    expect(def.valueType).toBeInstanceOf(StringDefinition);
  });
});
