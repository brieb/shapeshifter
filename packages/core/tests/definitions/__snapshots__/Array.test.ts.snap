// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`definitions/Array errors if \`valueType\` is empty 1`] = `
"Invalid ArrayDefinition field \\"valueType\\". Type must be one of: string, shape. Received string with the following invalidations:
 - String cannot be empty."
`;
